evmcore/tx_validation.go

```diff
@@ -125,19 +116,73 @@ func validateTx(tx *types.Transaction, opt validationOptions) error {
 		return ErrTipAboveFeeCap
 	}
 
-	// Make sure the transaction is signed properly.
-	from, err := types.Sender(opt.signer, tx)
-	if err != nil {
-		return ErrInvalidSender
+	// Sonic only supports Blob transactions without blob data.
+	if tx.Type() == types.BlobTxType &&
+		(len(tx.BlobHashes()) > 0 || (tx.BlobTxSidecar() != nil && len(tx.BlobTxSidecar().BlobHashes()) > 0)) {
+		return ErrNonEmptyBlobTx
 	}
 
-	// Drop non-local transactions under our own minimal accepted gas price or tip
-	local := opt.isLocal || opt.locals.contains(from) // account may be local even if the transaction arrived from the network
-	if !local && tx.GasTipCapIntCmp(opt.minTip) < 0 {
-		log.Trace("Rejecting underpriced tx: pool.gasPrice", "pool.gasPrice",
-			opt.minTip, "tx.GasTipCap", tx.GasTipCap())
-		return ErrUnderpriced
+	// Check non-empty authorization list
+	if tx.Type() == types.SetCodeTxType && len(tx.SetCodeAuthorizations()) == 0 {
+		return ErrEmptyAuthorizations
+	}
+
+	// Reject transactions over defined size to prevent DoS attacks
+	if uint64(tx.Size()) > txMaxSize {
+		return ErrOversizedData
+	}
```

Comment:
This property is a limit imposed by the transaction pool, not by the specification of transactions. This makes it not independent of any context. This check should probably be part of `validateTxForPool`.